### PR TITLE
Infrastructure "Only run workflow actions for relevant changes"

### PR DESCRIPTION
**Pull request by int-ua**
#### Summary

SUMMARY: None

#### Purpose of change

Do not run complete builds unnecessarily.

#### Describe the solution

Used [`on.<push|pull_request>.paths`](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) in GitHub workflows and [`skip_commits`](https://www.appveyor.com/docs/how-to/filtering-commits/#skip-commits-2) in `appveyor.yml`  to try ignoring changes.

#### Describe alternatives you've considered

I'd like to use inclusion list instead of exclusion on the general build matrix but I need help with defining it.

#### Testing

By GitHub infrastructure.

#### Additional context

https://discord.com/channels/598523535169945603/598529174302490644/791280543496273941

### CODE REVIEW

**Comment by int-ua** [comment]
@BrettDong can you please take a look at translation template change sources and tell if they look ok?

File: .github/workflows/translations.yml [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
       - master
-name: "Push translation template"
+    paths:
+      - '**.json'
+      - '**.cpp'
+      - 'src/**'
+      - 'data/**'
+      - 'lang/extract_json_strings.py'
+      - 'lang/strip_line_numbers.py'
+      - 'lang/po/**.pot'
```
Comment (BrettDong) [inline .github/workflows/translations.yml:14]: I think this line is unnecessary now, the rest looks okay.

Comment (int-ua) [reply .github/workflows/translations.yml]: thanks, removed it. Now I think that having both `'**.json'` and `'data/**'` may be unnecessary, no? Same for `'**.cpp'` and `'src/**'`

Comment (BrettDong) [reply .github/workflows/translations.yml]: `**.cpp`, `**.h` and `**.json` is consistent with `xgettext ... src/*.cpp src/*.h lang/json/*.py` in `lang/update_pot.sh`

File: .github/workflows/astyle.yml [hunk 2]
```diff
@@ -1,7 +1,12 @@
 name: astyle
 
 on:
-  pull_request
+  pull_request:
+    paths:
+    - '**.json'
```
Comment (ZhilkinSerg) [inline .github/workflows/astyle.yml:6]: Probably not required

```suggestion
```

Comment (int-ua) [reply .github/workflows/astyle.yml:6]: Thank you for taking a look! The job in this file is called `style-json`, are you sure? Should the job be renamed?

Comment (actual-nh) [reply .github/workflows/astyle.yml:6]: I suggest renaming the job; it seems to entirely be doing astyle, not checking JSON.

Comment (int-ua) [reply .github/workflows/astyle.yml]: done, thanks

**Comment by ZhilkinSerg** [comment]
In order to make Github actions run faster we should probably decrease fetch depth too and do a shallow clone with depth of 1 instead of 10.

**Comment by int-ua** [comment]
Didn't notice that, thank you. Reduced in #46361

**Comment by int-ua** [comment]
Looks like @AppVeyor thinks that if build was skipped by the `skip_commits:` directive it's a failure :angry:

**Pull request merged**